src/components/Select/Select.tsx

```diff
@@ -0,0 +1,315 @@
+import {
+  type HTMLAttributes,
+  type Ref,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { cva } from "../../../styled-system/css";
+import { Icon } from "../Icon/Icon";
+
+export interface SelectProps extends HTMLAttributes<HTMLSelectElement> {
+  /** 선택된 값 */
+  value?: string;
+  /** 값이 변경될 때 호출되는 함수 */
+  onChange?: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  /** 비활성화 상태 */
+  disabled?: boolean;
+  /** 필수 입력 여부 */
+  required?: boolean;
+  /** 오류 상태 */
+  invalid?: boolean;
+  /** 지우기 버튼의 aria-label */
+  clearButtonName?: string;
+  /** placeholder 텍스트 */
+  placeholder?: string;
+  /** 역할을 설명하는 레이블 */
+  "aria-label"?: string;
+  /** DOM 요소 참조 */
+  ref?: Ref<HTMLSelectElement>;
+  /** 이름 */
+  name?: string;
+}
+
+/**
+ * - 네이티브 select 태그를 사용하는 Select 컴포넌트입니다.
+ * - 여러 선택지 중 하나를 고를 때 사용하며, 화면 공간을 절약하고 옵션 목록을 펼쳤을 때만 표시하고 싶을 때 적합합니다. 옵션이 6개 이상일 때 특히 유용합니다.
+ * - `children`으로 `<option>` 요소를 직접 전달할 수 있습니다.
+ * - `disabled` 속성으로 비활성화 상태를 제어할 수 있으며, `invalid` 속성을 통해 오류 상태를 표현할 수 있습니다.
+ * - `clearButtonName` 속성을 통해 선택된 값을 지울 수 있는 버튼을 표시할 수 있습니다.
+ *
+ * ### 접근성(Accessibility) 안내
+ * - label을 사용하거나 aria-label 속성을 설정하여 Select 컴포넌트의 역할을 명시적으로 설명해주세요.
+ * - invalid 속성을 true로 설정하면 오류 상태를 표시하며 aria-invalid 속성을 true로 설정합니다.
+ * - required 속성을 true로 설정하면 필수 입력 여부를 표시하며 aria-required 속성을 true로 설정합니다.
+ */
+export function Select({
+  value,
+  defaultValue,
+  disabled,
+  required,
+  invalid,
+  clearButtonName,
+  placeholder,
+  children,
+  onChange,
+  ref,
+  ...rest
+}: SelectProps) {
+  const selectRef = useRef<HTMLSelectElement>(null);
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const [hasValue, setHasValue] = useState<boolean>(!!(value || defaultValue));
+  const [overflowed, setOverflowed] = useState<boolean>(false);
+  const [titleText, setTitleText] = useState<string>("");
+
+  /**
+   * 내부 ref(overflowed 상태를 확인)와 외부에서 ref를 주입할 경우 덮어쓰기 방지를 위한 함수입니다.
+   * @param node - select 엘리먼트
+   */
+  const setRef = (node: HTMLSelectElement | null) => {
+    selectRef.current = node;
+    if (typeof ref === "function") {
+      ref(node);
+    } else if (ref) {
+      ref.current = node;
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (onChange) onChange(e);
+    setHasValue(!!e.target.value && e.target.value !== "");
```

Comment:
` e.target.value`가 빈 문자열일 때 && 연산자 좌측 값이 이미 false입니다. `e.target.value !== ""`는 체크할 필요가 없다고 생각해요.

```suggestion
    setHasValue(!!e.target.value);
```